apps/aws-amplify/src/locations/ConfigScreen.tsx

```diff
@@ -0,0 +1,156 @@
+import { useCallback, useState, useEffect, ChangeEvent } from 'react';
+import { ConfigAppSDK } from '@contentful/app-sdk';
+import {
+  Box,
+  Flex,
+  FormControl,
+  Heading,
+  HelpText,
+  Paragraph,
+  Stack,
+  TextInput,
+  TextLink,
+} from '@contentful/f36-components';
+import { css } from 'emotion';
+import { useSDK } from '@contentful/react-apps-toolkit';
+import tokens from '@contentful/f36-tokens';
+import { ExternalLinkIcon } from '@contentful/f36-icons';
+import AmplifyIcon, { AWSAmplifyBrand } from '../components/common/AmplifyIcon';
+import { isValidUrl } from '../lib/isVaildUrl';
+
+export interface AppInstallationParameters {
+  amplifyWebhookUrl?: string;
+}
+
+export const styles = {
+  body: css({
+    height: 'auto',
+    minHeight: '40vh',
+    margin: '0 auto',
+    marginTop: tokens.spacingXl,
+    padding: `${tokens.spacingXl} ${tokens.spacing2Xl}`,
+    maxWidth: '900px',
+    backgroundColor: tokens.colorWhite,
+    zIndex: 2,
+    boxShadow: '0px 0px 20px rgba(0, 0, 0, 0.1)',
+    borderRadius: '2px',
+  }),
+  background: css({
+    display: 'block',
+    position: 'absolute',
+    zIndex: -1,
+    top: 0,
+    width: '100%',
+    height: '300px',
+    backgroundColor: AWSAmplifyBrand.primaryColor,
+  }),
+  box: {
+    width: '100%',
+    marginTop: '22px',
+  },
+  icon: {
+    marginTop: '41px',
+  },
+};
+
+const ConfigScreen = () => {
+  const sdk = useSDK<ConfigAppSDK>();
+  const [parameters, setParameters] = useState<AppInstallationParameters>({});
+  const [amplifyWebhookUrl, setAmplifyWebhookUrl] = useState<string>('');
+
+  const onConfigure = useCallback(async () => {
+    const currentState = await sdk.app.getCurrentState();
+
+    if (!isValidUrl(amplifyWebhookUrl)) {
+      sdk.notifier.error('Please provide a valid webhook URL.');
+      return false;
+    }
+
+    const parametersToSave = {
+      ...parameters,
+      amplifyWebhookUrl,
+    };
+
+    setParameters(parametersToSave);
+
+    return {
+      parameters: parametersToSave,
+      targetState: currentState,
+    };
+  }, [parameters, sdk, amplifyWebhookUrl]);
+
+  useEffect(() => {
+    sdk.app.onConfigure(() => onConfigure());
+  }, [sdk, onConfigure]);
+
+  useEffect(() => {
+    (async () => {
+      const currentParameters: AppInstallationParameters | null = await sdk.app.getParameters();
+
+      if (currentParameters) {
+        setParameters(currentParameters);
+      }
+
+      if (currentParameters?.amplifyWebhookUrl) {
```

Comment:
I don't think it's necessary to do this check again, you can just set your amplifyWebhookUrl on line 91 since you're already checking for `currentParameters`.

Basically get rid of this if block and in the above just have
```
      if (currentParameters) {
        setParameters(currentParameters);
        setAmplifyWebhookUrl(currentParameters.amplifyWebhookUrl);
      }
```

Reply:
the TS compiler doesn't like the call though until explicitly checking the `AppInstallationParameters` field of `amplifyWebhookUrl` as a string, at that point it is still treating it as `SetStateAction<string>`

```typescript
Argument of type 'string | undefined' is not assignable to parameter of type 'SetStateAction<string>'.
  Type 'undefined' is not assignable to type 'SetStateAction<string>'
```

Can remove the default empty string from the `useState` call but then need to type that `as string` to overload the `string | undefined` which could potentially be missing on initial render. I feel like leaving it for now if we want to have a fast follow for cleanup of this?

Reply:
Wait, it complains? It works for me

```
  useEffect(() => {
    (async () => {
      const currentParameters = await sdk.app.getParameters();

      if (currentParameters) {
        setParameters(currentParameters);
        setAmplifyWebhookUrl(currentParameters.amplifyWebhookUrl);
      }

      sdk.app.setReady();
    })();
  }, [sdk]);
```